src/pages/index.js

```diff
@@ -81,35 +143,174 @@ profileEditButton.addEventListener("click", () => {
+);
+
+const editAvatarValidator = new FormValidator(
+  validationSettings,
+  document.querySelector("#edit-avatar-modal")
+);
 
 editFormValidator.enableValidation();
 addFormValidator.enableValidation();
-
+editAvatarValidator.enableValidation();
 /* -------------------------------------------------------------------------- */
 /*                                 Functions;                                 */
 /* -------------------------------------------------------------------------- */
-function handleImageClick(data) {
-  imagePopup.open(data);
+function handleCardPreview(cardData) {
+  previewImagePopup.open(cardData);
 }
+
+const avatarSubmitButton = document.querySelector("#avatar-submit-button");
+
+// function handleAvatarEditSubmit(input) {
+//   const link = input.avatar;
+//   api.updateAvatar(link);
+//   if (link) {
+//     avatarSubmitButton.textContent = "Saving...";
+//     api
+
+//       .editProfileImage(link)
+//       .then((userData) => {
+//         userInfo.setUserInfo(userData);
+//         editAvatarModal.close();
+//       })
+//       .catch((err) => console.error("Error updating avatar:", err))
+//       .finally(() => {
+//         editAvatarModal.setLoadingState(false);
+//       });
+//   } else {
+//     console.error("Avatar Link is not defined");
+//   }
+// }
+
+function handleAvatarEditSubmit(input) {
+  const link = input.avatar;
+
+  if (link) {
+    // set loading state to true
+    editAvatarModal.setLoadingState(true);
+    // the right api call is updateAvatar
+    api
+      .updateAvatar(link)
+      .then((userData) => {
+        // set avatar info
+        userInfo.setAvatarInfo(userData.avatar);
+
+        editAvatarModal.close();
+      })
+      .catch((err) => console.error("Error updating avatar:", err))
+      .finally(() => {
+        // set loading state to false
+        editAvatarModal.setLoadingState(false);
+      });
+  } else {
+    console.error("Avatar Link is not defined");
+  }
+}
+
 function handleProfileEditSubmit(formValues) {
   userInfo.setUserInfo({
     name: formValues.title,
     about: formValues.card__description,
   });
   editProfileModal.close();
 }
-function handleAddCardFormSubmit(formValues) {
-  const name = formValues.title;
-  const link = formValues.url;
-
-  const card = createCard({ name, link });
-  cardList.addItem(card);
-  addCardFormElement.reset();
-  addCardModal.close();
+// function handleAddCardFormSubmit(formValues) {
+//   const name = formValues.title;
+//   const link = formValues.url;
+
+//   const card = createCard({ name, link });
+//   cardList.addItem(card);
+//   addCardFormElement.reset();
+//   addCardModal.close();
+// }
+
+function handleDeleteClick(card) {
+  const cardId = card._id;
+  if (!cardId) {
+    console.error("card._id is undefined");
+    return;
+  }
+  confirmDeleteModal.setSubmitFunction(() => {
+    api
+      .deleteCard(cardId)
+      .then((res) => {
+        console.log(res);
+        card._handleDeleteCard();
```

Comment:
`_handleDeleteCard`

Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.